src/solver/conv_hip_implicit_gemm_bwd_v4r1_xdlops.cpp

```diff
@@ -420,27 +422,29 @@ bool PerformanceImplicitGemmBwdDataV4R1Xdlops::IsFastToBeUsedForTuning(
 
         const float ratio = float(grid_size) / grid_size_max_blockwise_gemm;
 
-        if(grid_size_max_blockwise_gemm > 600)
+        const auto num_cu = ctx.GetStream().GetMaxComputeUnits();
+
+        if(grid_size_max_blockwise_gemm > 5 * num_cu)
```

Comment:
I see this function if full of magic. Who is the magician :mage_man: that can fix this code in case of issues?

Reply:
@asroy Yes! 
```suggestion
        // This is to exclude some tuning parameters some result in very large
        // number (much larger than # of CUs) of workgroup from tuning.
        // This is just some heuristic. I guess the best thing I can do is to add comment here.
        // -- @asroy
        if(grid_size_max_blockwise_gemm > 5 * num_cu)
```

Reply:
@asroy And if would be nice to refactor this to have one copy of this code, see https://github.com/ROCmSoftwarePlatform/MIOpen/pull/553#discussion_r514253178

Reply:
@asroy Please provide some feedback here, thanks.

Reply:
Added comment. 

To take this logic out and put it in a separate function, I need to rework the logic a bit. I will do in a future PR

Reply:
Okay